File: QueueLinkedList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataStructures
{
    class LinkedListQueue
    {
        public class Node
        {
            public Node next;
            public Object data;
            public Node(Object data)
            {
                this.data = data;
            }
        }
        Node first,last;
        int size;
        public LinkedListQueue()
        {
            size = 0;
        }

        public void enqueue(Object data)
        {
            Node oldLast = last;
            last = new Node(data);
            last.next = null;
            if (size == 0)
            {
                first = last;
            }
            else
            {
                oldLast.next = last;
            }
            size++;
        }

        public Object dequeue()
        {
            Object deletedData = first.data;
            first = first.next;
            if(size == 0)
            {
                last = null;
            }
            size--;
            return deletedData;
        }
        public void printQueue()
        {
            Node current = first;
            while (current != null)
            {
                Console.Write("{0} ", current.data);
                current = current.next;
            }
            Console.WriteLine();
        }
        public Object peek()
        {
            return first.data;
        }
        public bool isEmpty()
        {
            if (size == 0)
            {
                Console.WriteLine("Queue is empty!");
                return true;
            }
            return false;
        }
        public int Size()
        {
            return size;
        }
    }
}

File: Java/Queues/QueueLinkedList.java
/*
 * This is a linked list implementation of a queue.
 * A queue is a FIFO data structure
 */
public class QueueLinkedList<T> {
	public class Node<T>{
		T data;
		Node next;
		Node(T data){
			this.data = data;
		}
	}
	//Instance Variables
	Node first, last;
	private int size;
	QueueLinkedList(){ //Constructor
		size = 0;
		first = null; last = null;
	}
	
	//Adds an element to the queue
	public void enqueue(T data){
		Node newNode = new Node(data);
		if(size == 0){
			newNode.next = first;
			first = newNode;
			last = first;
			size++;
			return;
		}
		else{
			last.next = newNode;
			last = newNode;
			size++;
		}
	}
	//Removes element from the queue
	public T dequeue(){
		if(isEmpty()){
			return null;
		}
		T dataToDequeue = (T)first.data;
		first = first.next;
		size--;
		return dataToDequeue;
	}
	public T peek(){ //Returns value of first element in queue
		return (T)first.data;
	}
	public void printQueue(){ //prints all values stored in the queue
		Node current = first;
		if(isEmpty()){
			return;
		}
		for(Node i = current; i != null; i = i.next){
			System.out.print(i.data + " ");
		}
		System.out.println();
	}
	/*
	 * Methods to check size of queue and if it is empty
	 */
	public Boolean isEmpty(){
		if(size == 0){
			System.out.println("Queue is empty!");
			return true;
		}
		return false;
	}
	public int getSize(){
		return size;
	}
}
